tfjs-backend-cpu/src/kernels/BatchMatMul.ts

```diff
@@ -106,12 +106,19 @@ export function batchMatMul(args: {
               let sum = 0.0;
 
               for (let k = k0; k < kBlock; k++) {
-                const batchOffsetA = Math.min(bi, batchDimA - 1) * aBatch;
-                const batchOffsetB = Math.min(bi, batchDimB - 1) * bBatch;
+                let batchIndexA = bi;
+                let batchIndexB = bi;
+                if (bi >= batchDimA) {
+                  batchIndexA = bi % batchDimA;
+                } else if (bi >= batchDimB) {
+                  batchIndexB = bi % batchDimB;
+                }
                 const aVal =
-                    a3dValues[batchOffsetA + i * aOuterStep + k * aInnerStep];
+                    // tslint:disable-next-line: max-line-length
+                    a3dValues[batchIndexA * aBatch + i * aOuterStep + k * aInnerStep];
                 const bVal =
-                    b3dValues[k * bInnerStep + j * bOuterStep + batchOffsetB];
+                    // tslint:disable-next-line: max-line-length
+                    b3dValues[k * bInnerStep + j * bOuterStep + batchIndexB * bBatch];
```

Comment:
I was wondering why `k` matched with `aInnerStep` but did not then match with `bOuterStep`, but I see now that `bOuterStep` and `bInnerStep` are opposite to `aOuterStep` and `aInnerStep` (lines 82 - 87). They don't refer to 'rows' and 'columns' of the matrices being multiplied, where you would step `a`'s row index with `b`'s column index for the dot product.

No action necessary.